cloud/services/container/clusters/reconcile.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusters
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/container/apiv1/containerpb"
+	"github.com/googleapis/gax-go/v2/apierror"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	infrav1exp "sigs.k8s.io/cluster-api-provider-gcp/exp/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Reconcile reconcile GKE cluster.
+func (s *Service) Reconcile(ctx context.Context) (ctrl.Result, error) {
+	log := log.FromContext(ctx).WithValues("service", "container.clusters")
+	log.Info("Reconciling cluster resources")
+
+	cluster, err := s.describeCluster(ctx)
+	if err != nil {
+		s.scope.GCPManagedControlPlane.Status.Ready = false
+		return ctrl.Result{}, err
+	}
+	if cluster == nil {
+		log.Info("Cluster not found, creating")
+		s.scope.GCPManagedControlPlane.Status.Ready = false
+		if err = s.createCluster(ctx); err != nil {
+			conditions.MarkFalse(s.scope.ConditionSetter(), infrav1exp.GKEControlPlaneReadyCondition, infrav1exp.GKEControlPlaneReconciliationFailedReason, clusterv1.ConditionSeverityError, err.Error())
+			conditions.MarkFalse(s.scope.ConditionSetter(), infrav1exp.GKEControlPlaneCreatingCondition, infrav1exp.GKEControlPlaneReconciliationFailedReason, clusterv1.ConditionSeverityError, err.Error())
+			return ctrl.Result{}, err
+		}
+		log.Info("Cluster provisioning in progress")
+		conditions.MarkFalse(s.scope.ConditionSetter(), infrav1exp.GKEControlPlaneReadyCondition, infrav1exp.GKEControlPlaneCreatingReason, clusterv1.ConditionSeverityInfo, "")
+		conditions.MarkTrue(s.scope.ConditionSetter(), infrav1exp.GKEControlPlaneCreatingCondition)
+		return ctrl.Result{RequeueAfter: 5 * time.Minute}, nil
```

Comment:
Would be good to extract this requeue period as a constant.

We could consider making it configurable via a flag, but perhaps that's overkill at this stage. wdyt?

Reply:
yea i was thinking about making it a constant but somehow forgot, it's now updated. 
we could make it configurable, but maybe let's start with a constant?

Reply:
Agreed, lets start with a constant. And thanks 👍